core/src/test/java/org/infinispan/partitionhandling/PreferConsistencyStrategyTest.java

```diff
@@ -45,5 +77,49 @@ public void testAvoidingNullPointerExceptionWhenUpdatingPartitionWithNullTopolog
       Assert.assertEquals(RebalancingStatus.COMPLETE, status.getRebalancingStatus());
    }
 
+   public void testCoordinatorHasNewerTopology() throws Exception {
+      Address myAddress = new TestAddress(1);
+      Address otherAddress = new TestAddress(2);
 
+      manager.addPersistentAddressMapping(myAddress, PersistentUUID.randomUUID());
+      manager.addPersistentAddressMapping(otherAddress, PersistentUUID.randomUUID());
+
+      ClusterCacheStatus coordinatorStatus = new ClusterCacheStatus("does-not-matter", preferConsistencyStrategy, topologyManager, null, Optional.empty(), manager);
+      coordinatorStatus.doJoin(myAddress, createJoinInfo(myAddress));
+      coordinatorStatus.doJoin(otherAddress, createJoinInfo(otherAddress));
+      // at this stage current topology ID from CacheStatus is 2
+
+      // The cache response contains non-null stable topology but null current topology
+      CacheStatusResponse responseFromOtherNode = new CacheStatusResponse(null,
+            null,
+            createCacheTopology(1, myAddress, otherAddress),
+            AvailabilityMode.AVAILABLE);
+
+      //when
+      preferConsistencyStrategy.onPartitionMerge(coordinatorStatus, Arrays.asList(responseFromOtherNode));
```

Comment:
I was trying to reproduce scenario mentioned in your [comment](https://github.com/infinispan/infinispan/pull/4871#discussion_r102774407): 

> I'm not sure about this change, I'm thinking maybe the coordinator already has a non-null topology in the ClusterCacheStatus and in that case, it should be replaced with a null topology.

And in this part:

> I think the idea was to create an initial topology if the cache has no "full" members yet, but there is one node who sent a join request to the old coordinator and didn't get a response with the initial cache topology before sending its status response. The joiner should be retrying to join with the new coordinator though, so I'm not sure it's worth keeping track of these joiners during the recovery process.
The case I was thinking about before is when there is a merge, but a node also left the cluster in the same view. If that node was the last member of one cache, we should make sure that cache's topology is set to null. Unfortunately you're right, we were trying to create an initial topology here too, and that probably caused a NPE sooner or later...

So in this case I'm setting current and stable topology to `null`. `CacheStatus` has newer topology (id=2) than `CacheStatusResponse` (id=1). So what you think? Is this scenario OK?